cam/cam.go

```diff
@@ -270,39 +270,54 @@ func (vs *videostore) DoCommand(_ context.Context, command map[string]interface{
 	// The response contains the name of the uploaded file.
 	case "save":
 		vs.logger.Debug("save command received")
-		from, to, metadata, err := validateSaveCommand(command)
+		from, to, metadata, async, err := validateSaveCommand(command)
 		if err != nil {
 			return nil, err
 		}
-		uploadFilePath, err := vs.conc.concat(from, to, metadata, vs.uploadPath)
-		if err != nil {
-			vs.logger.Error("failed to concat files ", err)
-			return nil, err
-		}
+		uploadFilePath := generateOutputFilePath(vs.name.Name, formatDateTimeToString(from), metadata, vs.uploadPath)
 		uploadFileName := filepath.Base(uploadFilePath)
-		return map[string]interface{}{
-			"command":  "save",
-			"filename": uploadFileName,
-		}, nil
+		switch async {
+		case true:
+			vs.logger.Debug("running save command asynchronously")
+			vs.workers.Add(func(ctx context.Context) {
+				vs.asyncSave(ctx, from, to, uploadFilePath)
```

Comment:
Things to consider:
1 - We add a worker with each DoCommand request, do we want them to interrupt already working workers?
2 - Should another call interrupt this worker?
3 - `workers.Stop` will stop all workers so if you want them to be separate from your other workers, give this new set of workers  a new variable in the struct.

Reply:
> 1 - We add a worker with each DoCommand request, do we want them to interrupt already working workers?

I do not want the do_command workers to interfere with the other long-running workers (processFrames, deleter). Adding workers should not interfere:
```
type StoppableWorkers struct {
	// Use a `sync.RWMutex` isntead of a `sync.Mutex` so that additions of new
	// workers do not lock with each other in any way. We want
	// as-fast-as-possible worker addition.
	mu         sync.RWMutex
	ctx        context.Context
	cancelFunc func()

	workers sync.WaitGroup
}
```

> 2 - Should another call interrupt this worker?

The short-lived asynSave routine should just die out when finished through the cancel call.

> 3. workers.Stop will stop all workers so if you want them to be separate from your other workers, give this new set of workers a new variable in the struct.

It is ok to stop all workers during close - processFrames, deleter, and all outstanding async calls can be shutdown together.